src/cryptoadvance/specter/util/fee_estimation.py

```diff
@@ -1,40 +1,115 @@
 import logging
+import re
+import requests
+import urllib3
+from requests.exceptions import ConnectionError
 
 logger = logging.getLogger(__name__)
 
 
+class FeeEstimationResult:
+    """A tiny object to pass the Fee Estimation Results around"""
+
+    def __init__(self, result):
+        self.result = result
+        self._error_message = None
+
+    @property
+    def error_message(self):
+        return self._error_message
+
+    @error_message.setter
+    def error_message(self, value):
+        if self._error_message != None:
+            self._error_message = self._error_message + " \n" + value
+        else:
+            self._error_message = value
+
+    @property
+    def result(self):
+        return self._result
+
+    @result.setter
+    def result(self, value):
+        self._result = value
+
+
 def get_fees(specter, config):
+    try:
+        return _get_fees(specter, config)
+    except Exception as e:
+        fee_estimation = FeeEstimationResult(
+            {
+                "fastestFee": 1,
+                "halfHourFee": 1,
+                "hourFee": 1,
+                "minimumFee": 1,
+            }
+        )
+        fee_estimation.error_message = f"Failed to Fetch fee estimation. Please use manual fee calculation, check the logs for details. Error: {e}"
+        logger.exception(e)
+        return fee_estimation
+
+
+def _get_fees(specter, config):
+    fee_estimation_result = FeeEstimationResult(None)  # Just in case we have errors
```

Comment:
Maybe better to pass here some defaults instead of `None`, like:
```
{
                "fastestFee": 1,
                "halfHourFee": 1,
                "hourFee": 1,
                "minimumFee": 1,
}
```

Otherwise there is no guarantee that the result will have `["hourFee"]` or other fees, that will cause exceptions at higher level.

Reply:
Ah, nevermind, I see you catch exception above anyway,

Reply:
I think you're right. If Core doesn't deliver anything useful, then it has reasonable error-messages but not reasonable default results. In that case, no exception is thrown.

Usually i try to avoid catching pure Exceptions AT ALL but in that case, external calls are done and we're very sure it's worth to recover from there (+ we do `logger.exception(e)` ).